internal/backend/runtime/omni/controllers/omni/omni.go

```diff
@@ -97,6 +118,10 @@ func (rt *resourceTracker) cleanup(ctx context.Context) error {
 		}
 	}
 
+	if destroyReady && opts.destroyReadyCallback != nil {
+		return opts.destroyReadyCallback()
+	}
```

Comment:
This runs only once if all items are destroy-ready right? It is a bit confusing this way imo, as it is passed differently (via opts) and behaves differently from `beforeDestroyCallback`.

Instead maybe we can return a `cleanupResult` struct from this `cleanup` function, containing the actually destroyed IDs for example, and on the caller site we can check and act on that. Wdyt?

Reply:
I wanted to keep the API without changes. Also most of the other places don't need this `cleanupResult`